File: code/594.最长和谐子序列.go
/*
 * @lc app=leetcode.cn id=594 lang=golang
 *
 * [594] 最长和谐子序列
 */

// @lc code=start
func findLHS(nums []int) (ans int) {
	sort.Ints(nums)
	begin := 0
	for end, num := range nums {
		for num-nums[begin] > 1 {
			begin++
		}
		if num-nums[begin] == 1 && end-begin+1 > ans {
			ans = end - begin + 1
		}
	}
	return
}

// @lc code=end

File: code/594.最长和谐子序列.java
/*
* @lc app=leetcode.cn id=594 lang=java
 *
 * [594] 最长和谐子序列
 */

// @lc code=start
class Solution {
    public int findLHS(int[] nums) {
        Arrays.sort(nums);
        int begin = 0, res = 0;
        for (int end=0; end<nums.length; end++) {
            while(nums[end]-nums[begin]>1)  begin++;
            if (nums[end]-nums[begin] == 1) {
                res = Math.max(end-begin+1, res);
            }
        }
        return res;
    }
}
// @lc code=end

